draft-ietf-httpbis-http2-secondary-certs.md

```diff
@@ -758,6 +758,34 @@ any stream which references the failing certificate in question or process the
 requests as unauthenticated and provide error information at the HTTP semantic
 layer.
 
+# Required Domain Certificate Extension {#extension}
+
+The Required Domain extension allows certificates to limit their use with
+Secondary Certificate Authentication.  A client MUST verify that the server has
+proven ownership of the indicated identity before accepting the limited
+certificate over Secondary Certificate Authentication.
+
+The identity in this extension is a restriction asserted by the requester of the
+certificate and is not verified by the CA.  Conforming CAs SHOULD mark the
+requiredDomain extension as non-critical.  Conforming CAs MUST require the
+presence of a CAA record {{!RFC6844}} prior to issuing a certificate with this
+extension.
+
+The required domain is represented as a GeneralName, as specified in Section
+4.2.1.6 of {{!RFC5280}}. Unlike the subject field, conforming CAs MUST NOT issue
+certificates with a requiredDomain extension containing empty GeneralName
+fields.  Clients that encounter such a certificate when processing a
+certification path MUST consider the certificate invalid.
+
+The wildcard character "*" MAY be used to represent that any previously
+authenticated identity is acceptable.  This character MUST be the entirety of
+the name if used and MUST have a type of "dNSName".  (That is, "*" is
+acceptable, but "*.com" and "w*.example.com" are not).
```

Comment:
We discussed having stricter validation for certificates that use "\*".  It's not appropriate to mandate that, but this could mention that there are far greater risks to issuing certificates with a required domain of "\*".